### PR TITLE
Added macros to core plugin batch sources

### PR DESCRIPTION
**Pull request by knazir**
Enabled macros on the following properties:

`AzureBatchSource(s)`:
- account
- container
- storageKey

`BatchReadableSource(s)`:
- name

`ExcelInputReader`:
- filePath
- filePattern
- memoryTableName
- tableExpiryPeriod
- reprocess
- sheet
- sheetValue
- skipFirstRow
- terminateIfEmptyRow
- rowsLimit
- ifErrorRecord
- errorDatasetName

`FTPBatchSource`:
- path
- fileSystemProperties
- fileRegex
- inputFormatClassName

`FileBatchSource`:
- path
- fileSystemProperties
- fileRegex
- timeTable
- inputFormatClass
- maxSplitSize

`KVTableSource`
- name

`S3BatchSource(s)`:
- accessID
- accessKey

`SnapshotFileBatchSource(s)`:
- name
- basePath
- fileProperties

`TimePartitionedFileSetSource(s)`:
- name
- basePath
- duration
- delay

`XMLReaderBatchSource`:
- path
- nodePath
- pattern
- actionAfterProcess
- targetFolder
- reprocessingRequired
- tableName
- tableExpiryPeriod
- temporaryFolder

`TableSource`:
- rowField

### CODE REVIEW

**Comment by knazir** [comment]
Build running: http://builds.cask.co/browse/HYP-BAD54-4

File: core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/BatchReadableSource.java [hunk 1]
```diff
@@ -37,18 +39,28 @@
...
+    this.batchReadableWritableConfig = batchReadableWritableConfig;
+  }
 
   @Override
   public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
-    String name = getProperties().get(Properties.BatchReadableWritable.NAME);
-    Preconditions.checkArgument(name != null && !name.isEmpty(), "Dataset name must be given.");
-    String type = getProperties().get(Properties.BatchReadableWritable.TYPE);
-    Preconditions.checkArgument(type != null && !type.isEmpty(), "Dataset type must be given.");
+    // null check for tests, as macro fields map and properties are null in test cases
```
Comment (albertshau) [inline core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/BatchReadableSource.java]: what does this comment mean?

File: core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/BatchReadableSource.java [hunk 2]
```diff
@@ -37,18 +39,28 @@
...
+    if (!batchReadableWritableConfig.containsMacro(Properties.BatchReadableWritable.NAME)) {
+      String datasetName = getProperties().get(Properties.BatchReadableWritable.NAME);
```
Comment (albertshau) [inline core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/BatchReadableSource.java]: not from your PR, but I'm not sure why we do getProperties().get(name). Should be able to just use batchReadableWritableConfig.getName().

File: core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/BatchReadableSource.java (hunk 1 above)
Comment (knazir) [inline core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/BatchReadableSource.java]: Was left over from when there was a null check for `getProperties()`, forgot to remove it.

**Comment by knazir** [comment]
Guild is breen: http://builds.cask.co/browse/HYP-BAD54-4

File: core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/BatchReadableSource.java [hunk 3]
```diff
@@ -47,11 +47,10 @@ protected BatchReadableSource(BatchReadableWritableConfig batchReadableWritableC
 
   @Override
   public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
-    // null check for tests, as macro fields map and properties are null in test cases
     if (!batchReadableWritableConfig.containsMacro(Properties.BatchReadableWritable.NAME)) {
       String datasetName = getProperties().get(Properties.BatchReadableWritable.NAME);
       Preconditions.checkArgument(datasetName != null && !datasetName.isEmpty(), "Dataset name must be given.");
-      String datasetType = getProperties().get(Properties.BatchReadableWritable.TYPE);
+      String datasetType = batchReadableWritableConfig.getName();
```
Comment (albertshau) [inline core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/BatchReadableSource.java]: this should be for the name a couple lines up and not for the type.

**Comment by albertshau** [comment]
one comment

File: core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/BatchReadableSource.java (hunk 3 above)
Comment (knazir) [inline core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/BatchReadableSource.java]: ah crap sorry, brain turned off.. thanks

**Comment by russorat** [comment]
@knazir why so few fields on the S3 Plugins? Can we enable the other fields so that it matches the File source?

**Comment by russorat** [comment]
@knazir also, assuming we are working on enabling macro fields in the other sets of hydrator plugins?

**Comment by knazir** [comment]
The guild is breen: http://builds.cask.co/browse/HYP-BAD54-7

**Comment by knazir** [comment]
@russorat yep, there were 4 PRs merged earlier today for macros for other plugins, I'll look into the S3 plugins:
- https://github.com/caskdata/hydrator-plugins/pull/398
- https://github.com/caskdata/hydrator-plugins/pull/396
- https://github.com/caskdata/hydrator-plugins/pull/395
- https://github.com/caskdata/hydrator-plugins/pull/394

**Comment by albertshau** [comment]
lgtm

**Comment by knazir** [comment]
@russorat my bad, I forgot to update the documentation, `S3BatchSource` which extends `FileBatchSource` already has the fields enabled.

**Pull request merged**